modin/pandas/base.py

```diff
@@ -49,6 +49,7 @@
 from typing import Optional, Union, Sequence, Hashable
 import warnings
 import pickle as pkl
+from functools import cached_property
```

Comment:
Do you think what we can do with this?

Reply:
Is there a reason to _not_ use the same-looking `@cache_readonly` from pandas? The issue with `@cached_property` from `functools` is that it makes an _assignable_ attribute after the first call, which might not be desired.

Reply:
The only potential reason not to use cache_readonly is that it isn't public.  it isn't going anywhere though, so that doesn't bother me.  will change.

Reply:
Does ‘@cache_readonly’ work for python 3.6 and newer?

Reply:
yes